Remotion/Web/Development.WebTesting/WebTestSetUpFixtureHelper.cs

```diff
@@ -107,6 +117,51 @@ private void HostWebApplication ()
       _hostingStrategy.DeployAndStartWebApplication();
     }
 
+    private void VerifyWebApplicationStarted (string webApplicationRoot, TimeSpan applicationPingTimeout)
+    {
+      var stopwatch = Stopwatch.StartNew();
+      
+      var webRequest = (HttpWebRequest)HttpWebRequest.Create (webApplicationRoot);
+      webRequest.Method = WebRequestMethods.Http.Head;
+      webRequest.AllowAutoRedirect = true;
+      
+      //Just setting this to some value for the first run
+      var statusCode = HttpStatusCode.NotFound;
+
+      while (statusCode != HttpStatusCode.OK)
+      {
+        try
+        {
+          var response = (HttpWebResponse) webRequest.GetResponse();
+          statusCode = response.StatusCode;
+
+          if (stopwatch.ElapsedMilliseconds > applicationPingTimeout.TotalMilliseconds)
+          {
+            throw new InvalidOperationException (

```

Comment:
Throwing the `InvalidOperationException` here might not be the best design, as it gets caught and rethrown through the catch block, resulting in a weirdly composed exception message.

Reply:
You are completely right. I will move this Code block out of the try block